validator/src/machine/signing/sign.ts

```diff
@@ -0,0 +1,137 @@
+import { signRequestEventSchema } from "../../consensus/schemas.js";
+import type { SigningClient } from "../../consensus/signing/client.js";
+import { decodeSequence } from "../../consensus/signing/nonces.js";
+import type { VerificationEngine } from "../../consensus/verify/engine.js";
+import type {
+	ConsensusState,
+	MachineConfig,
+	MachineStates,
+	StateDiff,
+} from "../types.js";
+
+const NONCE_THRESHOLD = 100n;
+
+export const handleSign = async (
+	machineConfig: MachineConfig,
+	verificationEngine: VerificationEngine,
+	signingClient: SigningClient,
```

Comment:
Again, not for this PR, but one thing that caught my attention is that the only thing that is "side-effecty" right now are the clients. This can make a state transition valid or invalid depending on the internal state of the client (which is the only thing holding back making state transitions 100% replayable). Just some food for thought (as the "side-effecty-ness" is reasonable in this context IMO, and a valid optimization).

Reply:
I do think it is fair to refactor this and would start this after we setup the internal testenet (Note: I will create a project for this)

Reply:
https://www.notion.so/safefoundation/Shieldnet-Stateless-key-gen-and-signing-client-2bda8a34f3b8809491c0faaa1ddfe1fc?source=copy_link